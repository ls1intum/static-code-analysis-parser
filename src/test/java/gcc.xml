<?xml version="1.0" encoding="UTF-8" ?>
<root>
        ascii_table.c: In function ‘power’:
        ascii_table.c:7:13: warning: variable ‘arr’ set but not used [-Wunused-but-set-variable]
        7 |         int arr[5];
        |             ^~~
        ascii_table.c: In function ‘decimal_to_hexadecimal’:
        ascii_table.c:32:15: warning: implicit declaration of function ‘malloc’ [-Wimplicit-function-declaration]
        32 |         ptr = malloc(sizeof(int));
        |               ^~~~~~
        ascii_table.c:2:1: note: include ‘&lt;stdlib.h&gt;’ or provide a declaration of ‘malloc’
        1 | #include &lt;stdio.h&gt;
        +++ |+#include &lt;stdlib.h&gt;
        2 |
        ascii_table.c:32:15: warning: incompatible implicit declaration of built-in function ‘malloc’ [-Wbuiltin-declaration-mismatch]
        32 |         ptr = malloc(sizeof(int));
        |               ^~~~~~
        ascii_table.c:32:15: note: include ‘&lt;stdlib.h&gt;’ or provide a declaration of ‘malloc’
        ascii_table.c:31:14: warning: variable ‘ptr’ set but not used [-Wunused-but-set-variable]
        31 |         int *ptr;
        |              ^~~
        ascii_table.c:40:16: warning: leak of ‘ptr’ [CWE-401] [-Wanalyzer-malloc-leak]
        40 |         return hex_num;
        |                ^~~~~~~
        ‘decimal_to_hexadecimal’: events 1-4
        |
        |   32 |         ptr = malloc(sizeof(int));
        |      |               ^~~~~~~~~~~~~~~~~~~
        |      |               |
        |      |               (1) allocated here
        |   33 |
        |   34 |         for (int i = 0; dec_num != 0; i++) {
        |      |                         ~~~~~~~~~~~~
        |      |                                 |
        |      |                                 (2) following ‘false’ branch (when ‘dec_num == 0’)...
        |......
        |   40 |         return hex_num;
        |      |                ~~~~~~~
        |      |                |
        |      |                (3) ...to here
        |      |                (4) ‘ptr’ leaks here; was allocated at (1)
        |
</root>