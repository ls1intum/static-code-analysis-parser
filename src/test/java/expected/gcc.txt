{"tool":"GCC","issues":[{"filePath":"ascii_table.c","startLine":7,"endLine":7,"startColumn":13,"endColumn":13,"rule":"[-Wunused-but-set-variable]","category":"Misc","message":"[-Wunused-but-set-variable]: variable ‘arr’ set but not used \n        7 |         int arr[5];\n        |             ^~~\n        ascii_table.c: In function ‘decimal_to_hexadecimal’:","priority":"warning"},{"filePath":"ascii_table.c","startLine":32,"endLine":32,"startColumn":15,"endColumn":15,"rule":"[-Wimplicit-function-declaration]","category":"Misc","message":"[-Wimplicit-function-declaration]: implicit declaration of function ‘malloc’ \n        32 |         ptr = malloc(sizeof(int));\n        |               ^~~~~~","priority":"warning"},{"filePath":"ascii_table.c","startLine":32,"endLine":32,"startColumn":15,"endColumn":15,"rule":"[-Wbuiltin-declaration-mismatch]","category":"Misc","message":"[-Wbuiltin-declaration-mismatch]: incompatible implicit declaration of built-in function ‘malloc’ \n        32 |         ptr = malloc(sizeof(int));\n        |               ^~~~~~","priority":"warning"},{"filePath":"ascii_table.c","startLine":31,"endLine":31,"startColumn":14,"endColumn":14,"rule":"[-Wunused-but-set-variable]","category":"Misc","message":"[-Wunused-but-set-variable]: variable ‘ptr’ set but not used \n        31 |         int *ptr;\n        |              ^~~","priority":"warning"},{"filePath":"ascii_table.c","startLine":40,"endLine":40,"startColumn":16,"endColumn":16,"rule":"[-Wanalyzer-malloc-leak]","category":"Memory","message":"[-Wanalyzer-malloc-leak]: leak of ‘ptr’ [CWE-401] \n        40 |         return hex_num;\n        |                ^~~~~~~\n        ‘decimal_to_hexadecimal’: events 1-4\n        |\n        |   32 |         ptr = malloc(sizeof(int));\n        |      |               ^~~~~~~~~~~~~~~~~~~\n        |      |               |\n        |      |               (1) allocated here\n        |   33 |\n        |   34 |         for (int i = 0; dec_num != 0; i++) {\n        |      |                         ~~~~~~~~~~~~\n        |      |                                 |\n        |      |                                 (2) following ‘false’ branch (when ‘dec_num == 0’)...\n        |......\n        |   40 |         return hex_num;\n        |      |                ~~~~~~~\n        |      |                |\n        |      |                (3) ...to here\n        |      |                (4) ‘ptr’ leaks here; was allocated at (1)\n        |\n","priority":"warning"}]}
